core/src/main/kotlin/com/expediagroup/sdk/core/client/OkHttpEventListener.kt

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (C) 2022 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package main.kotlin.com.expediagroup.sdk.core.client
+
+import com.expediagroup.sdk.core.constant.HeaderKey
+import com.expediagroup.sdk.core.plugin.logging.ExpediaGroupLoggerFactory
+import okhttp3.Call
+import okhttp3.Connection
+import okhttp3.EventListener
+import okhttp3.Handshake
+import okhttp3.Protocol
+import okhttp3.Request
+import okhttp3.Response
+import java.io.IOException
+import java.net.InetSocketAddress
+import java.net.Proxy
+
+object OkHttpEventListener : EventListener() {
+    private val log = ExpediaGroupLoggerFactory.getLogger(this::class.java)
+
+    override fun callStart(call: Call) {
+        super.callStart(call)
+        log.debug("call start for Transaction-Id: %s".format(call.request().headers.get(HeaderKey.TRANSACTION_ID)))
```

Comment:
Can we capitalize the first word of the log messages? Also, we can use brackets like `"Call start for request with transaction-id: [%s] ..."` to better present the transaction id.